src/temperature_component.cpp

```diff
@@ -431,51 +426,58 @@ void TemperatureComponent::run( const double runToDate ) {
         DelQO = QO[tstep] - QO[tstep - 1];
 
         // Assume linear forcing change between tstep and tstep+1
-        QC1 = (DelQL/cal*(1.0/taucfl+1.0/taukls)-bsi*DelQO/cas/taukls);
-        QC2 = (DelQO/cas*(1.0/taucfs+bsi/tauksl)-DelQL/cal/tauksl);
-        QC1 = QC1 * pow(dt, 2.0)/12.0;
-        QC2 = QC2 * pow(dt, 2.0)/12.0;
+        QC1 = (DelQL / cal * (1.0 / taucfl + 1.0 / taukls) - bsi * DelQO / cas / taukls);
+        QC2 = (DelQO / cas * (1.0 / taucfs + bsi / tauksl) - DelQL / cal / tauksl);
+        QC1 = QC1 * pow(dt, 2.0) / 12.0;
+        QC2 = QC2 * pow(dt, 2.0) / 12.0;
 
         // ----------------- Initial Conditions --------------------
         // Initialization of temperature and forcing vector:
         // Factor 1/2 in front of Q in Equation A.27, EK05, and Equation 2.3.27, TK07 is a typo!
         // Assumption: linear forcing change between n and n+1
-        DQ1 = 0.5*dt/cal*(QL[tstep]+QL[tstep-1]);
-        DQ2 = 0.5*dt/cas*(QO[tstep]+QO[tstep-1]);
+        DQ1 = 0.5 * dt / cal * (QL[tstep] + QL[tstep - 1]);
+        DQ2 = 0.5 * dt / cas * (QO[tstep] + QO[tstep - 1]);
         DQ1 = DQ1 + QC1;
         DQ2 = DQ2 + QC2;
 
         // ---------- SOLVE MODEL ------------------
         // Calculate temperatures
         for(int i = 0; i <= tstep; i++) {
-            DPAST2 = DPAST2 + temp_sst[i] * Ker[ns-tstep+i-1];
+            DPAST2 = DPAST2 + temp_sst[i] * Ker[ns - tstep + i - 1];
         }
-        DPAST2 = DPAST2 * fso * pow((dt/taudif), 0.5);
+        DPAST2 = DPAST2 * fso * pow((dt / taudif), 0.5);
 
-        DTEAUX1 = A[0] * temp_landair[tstep-1] + A[1] * temp_sst[tstep-1];
-        DTEAUX2 = A[2] * temp_landair[tstep-1] + A[3] * temp_sst[tstep-1];
+        DTEAUX1 = A[0] * temp_landair[tstep - 1] + A[1] * temp_sst[tstep - 1];
+        DTEAUX2 = A[2] * temp_landair[tstep - 1] + A[3] * temp_sst[tstep - 1];
 
         temp_landair[tstep] = IB[0] * (DQ1 + DPAST1 + DTEAUX1) + IB[1] * (DQ2 + DPAST2 + DTEAUX2);
         temp_sst[tstep] = IB[2] * (DQ1 + DPAST1 + DTEAUX1) + IB[3] * (DQ2 + DPAST2 + DTEAUX2);
-    }
-    else {  // Handle the initial conditions
+    } else {  // Handle the initial conditions
         temp_landair[0] = 0.0;
         temp_sst[0] = 0.0;
     }
     temp[tstep] = flnd * temp_landair[tstep] + (1.0 - flnd) * bsi * temp_sst[tstep];
 
+    // If the user has supplied temperature data, use that instead
+    if( tgav_constrain.size() && runToDate <= tgav_constrain.lastdate() ) {
+        H_LOG( logger, Logger::WARNING ) << "** Overwriting temperatures with user-supplied value" << std::endl;
+        temp[tstep] = tgav_constrain.get( runToDate );
```

Comment:
Add a comment here describing what is going on in `temp_landair ` and `temp_sst`.